samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+/* Simple program:  Create a native window and attach an SDL renderer */
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <hal/video.h>
+#include "stdio.h"
+#include "string.h"
+#include <SDL.h>
+#include <SDL_ttf.h>
+
+int _exit(int x)
+{
+    debugPrint("_exit called\n");
+    debugPrint( " SDL_Error: %s\n", SDL_GetError() );
+    XSleep(5000);
+    XReboot();
+    return 0;
+}
+
+//Screen dimension constants
+const extern int SCREEN_WIDTH;
+const extern int SCREEN_HEIGHT;
+#define NUM_SPRITES 10
+#define MAX_SPEED   5
+
+static SDL_Rect *positions, *velocities;
+
+/* Call this instead of exit(), so we can clean up SDL: atexit() is evil. */
+static void
+quit(int rc)
+{
+    SDL_VideoQuit();
+    _exit(rc);
+}
+
+SDL_Texture *
+LoadSprite(SDL_Renderer *renderer)
+{
+    SDL_Surface *temp;
+    SDL_Texture *sprite;
+    TTF_Font    *font;
+
+    SDL_Color    color = {0x7F, 0x7F, 0xFF, 0xFF};
+
+    font = TTF_OpenFont("pennaregular.ttf", 32);
+    if (font == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't load font: %s",
+                     SDL_GetError());
+        return NULL;
+    }
+
+    temp = TTF_RenderText_Blended(font, "NXDK", color);
+    if (temp == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create surface: %s", SDL_GetError());
+        return NULL;
+    }
+
+    TTF_CloseFont(font);
+    /* Create textures from the image */
+    sprite = SDL_CreateTextureFromSurface(renderer, temp);
+    if (sprite == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create texture: %s\n", SDL_GetError());
+        SDL_FreeSurface(temp);
+        return NULL;
+    }
+    SDL_FreeSurface(temp);
+
+    /* We're ready to roll. :) */
+    return sprite;
+}
+
+void
+MoveSprites(SDL_Renderer * renderer, SDL_Texture * sprite)
+{
+    int sprite_w, sprite_h;
+    int i;
+    SDL_Rect viewport;
+    SDL_Rect *position, *velocity;
+
+    /* Query the sizes */
+    SDL_RenderGetViewport(renderer, &viewport);
+    SDL_QueryTexture(sprite, NULL, NULL, &sprite_w, &sprite_h);
+
+    /* Draw a gray background */
+    SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
+    SDL_RenderClear(renderer);
+
+    /* Move the sprite, bounce at the wall, and draw */
+    for (i = 0; i < NUM_SPRITES; ++i) {
+        position = &positions[i];
+        velocity = &velocities[i];
+        position->x += velocity->x;
+        if ((position->x < 0) || (position->x >= (viewport.w - sprite_w))) {
+            velocity->x = -velocity->x;
+            position->x += velocity->x;
+        }
+        position->y += velocity->y;
+        if ((position->y < 0) || (position->y >= (viewport.h - sprite_h))) {
+            velocity->y = -velocity->y;
+            position->y += velocity->y;
+        }
```

Comment:
All of this logic just masquerades what this sample does.

If I look at an TTF sample, all I really care about how I can integrate text into my program as quickly as possible.
The code we supply should be re-usable so people can just open a sample and copy/paste everything they need (so we should reduce amount of garbage between useful parts), and strip it down.

I specifically don't care about a physics simulation.

- This should be removed.

If we do anything else, other than trivial text rendering, we should probably show various variations: italic, outlines, auto-linebreaks, alignments in a textbox, ... (although it's not critical for now).